File: week_07/day_1/Try_Catch_Exceptions/code/Try_Catch_End/app/src/main/java/example/codeclan/com/try_catch_end/PetShop.java
package example.codeclan.com.try_catch_end;

/**
 * Created by user on 30/08/2017.
 */

import java.util.ArrayList;

public class PetShop{

    private ArrayList<Pet> stock;

    public PetShop(){
        stock = new ArrayList<Pet>();
    }
    // Add the throws keyword to warn the dev who's using our method
    public Pet findPetByName(String searchName) throws NullStringException{
        // Add an if statement to check the input as soon as you get it.
        // If the input is null, we can't do anything with it
        // We can't even search for a pet with that name, as it's not a valid String
        // So we throw our Exception, to explain what's wrong to the dev using our function.
        if (searchName == null){
            throw new NullStringException("Cannot search for a pet with null instead of a name String");
        }

        String searchLower = searchName.toLowerCase();
        for (Pet pet : stock){
            String petName = pet.getName().toLowerCase();
            if (petName.equals(searchLower)){
                return pet;
            }
        }
        // returns null if the pet's not found but the input was valid.
        return null;
    }

    public void addPet(Pet pet){
        stock.add(pet);
    }

}

File: week_10/c_sharp_course/week_1/day_5/try_catch/code/start_code/PetShop.cs
using System.Collections.Generic;

public class PetShop
{
  private List<Pet> stock;

  public PetShop(){
    stock = new List<Pet>();
  }

  public Pet FindPetByName(string searchName) 
  {    
    string searchLower = searchName.ToLower();
    foreach (var pet in stock)
    {
      string petName = pet.GetName().ToLower();
      if ( petName.Equals(searchLower) ) 
      {
        return pet;
      }
    }
    return null;
  }

  public void AddPet(Pet pet)
  {
    stock.Add(pet);
  }

}